scripts/build/gen_isr_tables.py

```diff
@@ -131,12 +131,65 @@ def parse_args():
 #endif
 """
 
+sh_isr_function = "void __attribute__((section(\".shared_isr\")))shared_isr_"
+sh_isr_args = "const size_t shared_isr_"
+sh_isr_args_section = " __attribute__((section(\".shared_isr_args\")))"
```

Comment:
So, IIUC the sections are going to be populated only by this script. So before this script is run I would expect the sections to be empty. So, are you sure this is not going to modify (shifting) the addresses between the 1st and the 2nd linking stage?

Given that you provided no test I cannot verify it. Please, verify it by comparing the `linker_zephyr_pre{0,1}` files.

Reply:
No shifting occurs because I've placed the .shared_isr and .shared_isr_args sections at the end of all other sections.

I have some tests but I can't get the current tests,  [arm_irq_advanced_features](https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/arch/arm/arm_irq_advanced_features), to pass. Is this the proper way to run them: twister -T tests/arch/arm/arm_irq_advanced_features/

ERROR   - mps2_an521 tests/arch/arm/arm_irq_advanced_features/arch.arm.irq_advanced_features.secure_fw FAILED: Unknown
ERROR   - mps2_an521 tests/arch/arm/arm_irq_advanced_features/arch.arm.irq_advanced_features FAILED: Unknown
ERROR   - mps2_an385 tests/arch/arm/arm_irq_advanced_features/arch.arm.irq_advanced_features FAILED: Unknown
ERROR   - qemu_cortex_m0  tests/arch/arm/arm_irq_advanced_features/arch.arm.irq_advanced_features FAILED: Unknown
ERROR   - mps2_an521_remote  tests/arch/arm/arm_irq_advanced_features/arch.arm.irq_advanced_features FAILED: Unknown
INFO    - Total complete:   86/  86  100%  skipped:   80, failed:    5